src/utils/isNodeProcess.react-native.test.ts

```diff
@@ -0,0 +1,25 @@
+// @ts-nocheck
+/**
+ * Test in React Native Environment
+ */
+
+import { isNodeProcess } from './isNodeProcess'
+
+test('returns true when run in a react native environment', () => {
+  const originalProcess = global.process
+  global.process = {
+    env: {
+      NODE_ENV: 'development',
+    },
+  }
+
+  const originalNavigator = global.navigator
+  Object.defineProperty(global.navigator, 'product', {
+    get: () => {
+      return 'ReactNative'
+    },
+  })
+  expect(isNodeProcess()).toBe(true)
```

Comment:
If the assertion fails, we will never restore the original validator (the function will short-circuit).
I highly recommend to stub `global.navigator` in the `beforeAll` and restore it in `afterAll` hooks in Jest.

Reply:
yes, I should stub global.navigator in the beforeAll and restore it in afterAll hooks, which will not break tests

Reply:
I think not having `global.process` explicitly at all brings us closer to testing the React Native logic in isolation.

Let's say we've got an issue in our `isNodeProcess()` that produces false positives. Now, since we also stub `global.process`, it may fall into the first condition that checks the `global.process` and produce a false positive. 

I think we shouldn't touch `global` in this test. What do you think?